reflection_refraction.py

```diff
@@ -0,0 +1,150 @@
+from distutils.util import strtobool
+import math
+from matplotlib import pyplot as plt
+
+distance = 5
+
+# Introduction
+def intro():
+    print("\nReflection and refraction of light Calculator")
+    print("By Martin Lambov")
+    print("GitHub: https://github.com/5neepy")
+
+# Check if the equetion is posible
+def check_if_posible(alfa, n1, n2):
+    if alfa >= 90 or alfa <= 0:
+        print("\nAlfa can't be equal or more than 90 and can't be equal or less than 0.\n")
+        exit()
+    if n1 == n2:
+        print("\nIf n_1 = n_2 we don't have Reflection/Refraction of light.\n")
+        exit()
+
+def get_vars():
+    alfa = float(input("\nEnter the degrees of alfa: "))
+    check_if_posible(alfa, 0, 1)
+
+    print("\nEnter the indexes of refraction of usual objects: \n air = 1 \n water = 1.33 \n glass = 1.5")
+
+    n1 = float(input("\nn1: "))
+    n2 = float(input("n2: "))
+    check_if_posible(alfa, n1, n2)
+
+    return alfa, n1, n2
+
+def calculate_refl_angl(alfa, n1, n2):
+    alfa_prim = alfa
+    
+    # Check if we have total internal reflection
+    tot_intern_refl = lambda theta, n1, n2: theta >= 45 and n1 > n2
+    
+    if tot_intern_refl(alfa, n1, n2):
+        beta = 90
+    else:
+        beta = math.degrees(math.asin(n1*math.sin(math.radians(alfa))/n2))  # n1*sin(θ1) = n2*sin(θ2)
```

Comment:
I don't know why, but when I write:
*alfa = 30*
*n1 = 30*
--> *alfa = n1*
*n2 is arbitrary*
The  program is crashing. Those trigonometric functions are so dumb. __Pls help!!!__

Reply:
Thanks!